### ISSUE TITLE
Shouldn't ESC always go to commands mode?

### ISSUE THREAD

**Issue by mimooh** [philc/vimium#2529]
After I create new tab (t) with vimium the cursor it positioned in address bar and I am in insert mode. In firefox plugins (e.g. vimperator) I could leave to commands mode with ESC and pressing 't' would open another tab. Vimium doesn't exit with ESC - the 't' is inserted as a character in address bar. Can I enable ESC escaping somehow?

**Comment by smblott-github** [comment]
Once the cursor is in Chrome's omnibox, there's no programmatic way to get the focus back in the page.  `Tab` works, though; I think.

(But I recommend [this](https://chrome.google.com/webstore/detail/new-tab-url/njigpponciklokfkoddampoienefegcl).)

**Comment by mimooh** [comment]
Ok, thanks! I will do with tab instead of ESC.

-- 
Karol Kreński
tel. 517-302-055

**Comment by chrysn** [comment]
could this be implemented on firefox? the restriction against programmatic interaction with the omnibar doesn't seem to apply to the firefox' address box.

**Comment by lydell** [comment]
@chrysn As far as I know Firefox’s location bar has the same restrictions as Chrome’s.

**Comment by jonniewhite** [comment]
@lydell: didn't VimFX process the ESC key in the location bar though?

**Comment by lydell** [comment]
@jonniewhite Yes it did! That’s because VimFx is a _legacy_ extension. They could do anything. It is the _new_ extensions (WebExtensions) that have the same restrictions as Chrome.

**Comment by chrysn** [comment]
Is there any particular item in bugzilla where one can leave whatever "This issue affects me as well" markers they use to get this fixed? Searching around there gives a lot of almost-but-not-quite duplicates [up to 20 years old](https://bugzilla.mozilla.org/show_bug.cgi?id=57805), but I found none that's concrete and actionable to Firefox devs.

**Comment by gdh1995** [comment]
Another extension like https://chrome.google.com/webstore/detail/new-tab-url/njigpponciklokfkoddampoienefegcl / https://chrome.google.com/webstore/detail/newtab-adapter/cglpcedifkgalfdklahhcchnjepcckfn is needed.

**Comment by chrysn** [comment]
That would solve the concrete issue of the new tab having its focus in the address bar, but not the generic issue of this thread that a press of escape in *any* situation should bring them to command mode.

**Comment by gdh1995** [comment]
Then it's an impossible task for a plain web extension.



---Original---
